client/dashboard/sites/overview-performance-card/index.tsx

```diff
@@ -0,0 +1,171 @@
+	site,
+	performanceData,
+	desktopScore,
+	mobileScore,
+}: {
+	site: Site;
+	performanceData: UrlPerformanceInsights;
+	desktopScore: number;
+	mobileScore: number;
+} ) {
+	const worseScore = Math.min( desktopScore, mobileScore );
+
+	const status = getPerformanceStatus( worseScore );
+	const statusText = getPerformanceStatusText( status );
+
+	const report =
+		desktopScore < mobileScore
+			? ( performanceData.pagespeed.desktop as PerformanceReport )
+			: ( performanceData.pagespeed.mobile as PerformanceReport );
+
+	const timeSinceLastTest = useTimeSince( report.timestamp );
+
+	let intent;
+	if ( status === 'poor' ) {
+		intent = 'error' as const;
+	} else if ( status === 'neutral' ) {
+		intent = 'warning' as const;
+	} else {
+		intent = 'success' as const;
+	}
+
+	let description;
+	if ( status === 'good' ) {
+		description = sprintf(
+			/* translators: %s: time since last test run */
+			__( 'Tested %s.' ),
+			timeSinceLastTest
+		);
+	} else {
+		const recommendationCount = Object.keys( report.audits ).length;
+		description = sprintf(
+			// translators: %(days) is the number of days until the link expires.
+			_n( '%d recommendation available.', '%d recommendations available.', recommendationCount ),
+			recommendationCount
+		);
+	}
+
+	const device = desktopScore < mobileScore ? 'desktop' : 'mobile';
+
+	return (
+		<OverviewCard
+			{ ...CARD_PROPS }
+			heading={ statusText }
+			description={ description }
+			intent={ intent }
+			link={ getPerformanceUrl( site, device ) }
+		/>
+	);
+}
+
+function PerformanceCardContentWithTests( { site }: { site: Site } ) {
+	const { performanceData, desktopScore, mobileScore } = usePerformanceData( site.ID, site.URL );
+
+	if ( performanceData === undefined ) {
+		return <OverviewCard { ...CARD_PROPS } isLoading />;
+	}
+
+	if ( desktopScore === undefined || mobileScore === undefined ) {
+		return <PerformanceCardContentWithoutTests site={ site } />;
+	}
+
+	return (
+		<PerformanceCardContentWithFinishedTests
+			site={ site }
+			performanceData={ performanceData }
+			desktopScore={ desktopScore }
+			mobileScore={ mobileScore }
+		/>
+	);
+}
+
+function PerformanceCardContent( { site }: { site: Site } ) {
+	const { data: settings } = useQuery( siteSettingsQuery( site.ID ) );
+
+	if ( settings === undefined ) {
+		return <OverviewCard { ...CARD_PROPS } isLoading />;
+	}
+
+	if ( site.launch_status === 'unlaunched' ) {
+		return (
+			<OverviewCard
+				{ ...CARD_PROPS }
+				heading={ __( 'No results' ) }
+				description={ __( 'Launch your site to test performance.' ) }
+				disabled
+			/>
+		);
+	}
+
+	if ( site.is_coming_soon || site.is_private ) {
+		return (
+			<OverviewCard
+				{ ...CARD_PROPS }
+				heading={ __( 'No results' ) }
+				description={ __( 'Make your site public to test performance.' ) }
+				disabled
+			/>
+		);
+	}
+
+	if ( ! settings.wpcom_performance_report_url ) {
+		return <PerformanceCardContentWithoutTests site={ site } />;
```

Comment:
It looks like we can return the `OverviewCard` directly, since it doesn't require any additional data like the ones above.

Reply:
Cannot because `<PerformanceCardContentWithoutTests />` is used twice in this file. 😄